test/stackTrace.js

```diff
@@ -98,6 +104,56 @@ tap.test('parses function name from original stack for anonymous function', func
     });
 });
 
+tap.test('parses function info from original stack for Promise scenario', function (tt) {
```

Comment:
this test needs to be skipped when `typeof Promise !== 'function'`

Reply:
Wrapped both `Promise` tests with the feature-detection code from `test/promises/fail.js`.